packages/app/obojobo-document-engine/src/scripts/oboeditor/components/visual-editor.js

```diff
@@ -299,7 +302,7 @@ class VisualEditor extends React.Component {
 		// This mostly happens with MoreInfoBoxes and void nodes
 		if (this.editor.selection) this.editor.prevSelection = this.editor.selection
 
-		this.setState({ value, saved: false })
+		this.setState({ value, saved: false, saving: false })
```

Comment:
This causes issues when, for example, when `saving` = `true` but the user clicks somewhere on the document. Then onChange is fired and the 'Saving...' text goes away.

I think instead we should coalesce this into a single property with multiple states (since `saved` and `saving` could theoretically both be true but that's an invalid state). My first idea is that we could change this to `saveState` which could be set to `"idle"`, `"saving"`, `"save-successful"` and `"save-failed"` - but I'm open to discussion.

Reply:
A click is registered as an edit, so I think it makes sense to stop showing `Saving...`

I change the state to `saveState`. I agree that it is a lot more flexible.